src/views/SignIn/SignIn.js

```diff
@@ -1,61 +1,33 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Link } from 'react-router-dom';
 import { SignInForm } from '../../components/SignInForm';
 
-import gql from 'graphql-tag';
-import { useApolloClient, useMutation } from '@apollo/react-hooks';
-
+import useSignInMutation from '../../hooks/useSignInMutation';
 import { useHistory } from 'react-router-dom';
 
 import './signin.scss';
 
-const SIGNIN_MUTATION = gql`
-  mutation SignInUser($data: SigninInput!) {
-    signInUser(data: $data) {
-      user {
-        id
-        username
-        firstName
-        lastName
-      }
-      jwt
-      authError
-    }
-  }
-`;
-
 const SignIn = (props) => {
-  const client = useApolloClient();
+  const { error, loading, signInUser } = useSignInMutation();
+  const [mutationError, setMutationError] = useState(null);
   const history = useHistory();
-  const [signInUser, { data: userData }] = useMutation(SIGNIN_MUTATION, {
-    onCompleted: (mutationResponse) => {
-      const { user, jwt, authError } = mutationResponse.signInUser;
-      if (!authError) {
-        localStorage.setItem('currentUser', JSON.stringify(user));
-        localStorage.setItem('jwt', jwt);
-        client.writeData({
-          data: { currentUser: user, jwt: jwt },
-        });
-        history.push('/home');
-      }
-    },
-    onError: (mutationError) => {
-      console.log('Mutation Error:', mutationError);
-    },
-  });
 
-  const authError = !!userData ? userData.signInUser.authError : null;
+  const onSubmit = async (input) => {
+    const { authError } = await signInUser(input);
+    setMutationError(authError);
+    if (!authError) history.push('/home');
```

Comment:
I would put this conditional line before the `setMutationError(authError);`. That way if the condition is true, it returns and does nothing else. Less work, yay!
```suggestion
    if (!authError) history.push('/home');
    setMutationError(authError);
```